### DISCUSSION TITLE
Select all diagnostics?

### DISCUSSION THREAD

**Discussion by matoous** [helix-editor/helix#3935] [Q&A]
Hi!

I tried searching around and maybe I missed something but is there an easy way to select all diagnostics in a document please?

**Comment by CBenoit** [comment]
Hi,

Just to make sure: you want to add a new selection on all code segment for which a diagnostic is raised?
I'm pretty sure this is not a feature currently supported, you could open a feature request issue for that.

**Reply by the-mikedavis** [reply]
I think this could be covered by https://github.com/helix-editor/helix/issues/3007 if diagnostics were considered textobjects (currently they are not but it would be a small change to do so). Then it would be `%` to select the whole document and then whatever the keybind would be for that command, then `d` for diagnostics

**Reply by rcorre** [reply]
This would be nice to have. I just copy-pasted some c++ code from one file to another with a different namespace. `clangd` flagged a bunch of functions/arguments/etc. because they were missing a namespace prefix. The ability to select all diagnostics and add the prefix would have been nice.
